### DISCUSSION TITLE
Show current context(function/method/...)

### DISCUSSION THREAD

**Discussion by robsonpeixoto** [LazyVim/LazyVim#948] [Q&A] [answered]
I'm moving from AstroNvim to LazyVim, because LazyVim is amazing, but I'm missing some features :cry: and I'm a NeoVim noob :sob: 

![image](https://github.com/LazyVim/LazyVim/assets/124390/06855f9a-f07d-45d1-a40c-68d27bfca456)

The AstroNvim show the current context on the top and it is really useful when you are editing a really big method/function and mainly to edit yaml files.

How can it be configured on LazyVim?

**Comment by dpetka2001** [comment, marked as answer]
LazyVim uses `nvim-navic` that shows the same information as a lualine statusline component. If you want to have it on the top instead, you could configure lualine winbar section to include what you want. Something like
```lua
  {
    "nvim-lualine/lualine.nvim",
    opts = {
      winbar = {
        lualine_c = {
          {
            function()
              return require("nvim-navic").get_location()
            end,
            cond = function()
              return package.loaded["nvim-navic"] and require("nvim-navic").is_available()
            end,
          },
        },
      },
    },
  },
```
should work. You then have to configure navic's seperators, icons and stuff because LazyVim customizes that as well. See [here](https://www.lazyvim.org/plugins/ui#nvim-navic) in the docs.

**Reply by dpetka2001** [reply]
You could also try [barbecue.nvim](https://github.com/utilyre/barbecue.nvim) for a more ready-to-go plugin if you don't want to mess around by hand a lot. This one also provides mouse-click support that takes you directly to where you clicked.

**Reply by robsonpeixoto** [reply]
Thanks @dpetka2001 , `barbecue` is amazing!

Is there an easy way to remove `nvim-navic` from [lualine_c](https://github.com/LazyVim/LazyVim/blob/e9358d977a379e54d73dd29691d60f7a51ed6923/lua/lazyvim/plugins/ui.lua#L103-L119)?

**Reply by dpetka2001** [reply]
If you didn't modify anything in `lualine_c` by yourself, then per the [docs](https://www.lazyvim.org/plugins/ui#lualinenvim) `nvim-navic` is the last element. So a simple
```lua
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      table.remove(opts.sections.lualine_c)
    end,
  },
```
should remove it. Otherwise if you added something else yourself, you should remove it by calculating the index (in current default LazyVim configuration it is index 4).
